features/netsocket/SocketAddress.cpp

```diff
@@ -215,17 +215,19 @@ void SocketAddress::set_port(uint16_t port)
 
 const char *SocketAddress::get_ip_address() const
 {
-    char *ip_address = (char *)_ip_address;
+    if (_addr.version == NSAPI_UNSPEC) {
```

Comment:
This does have the benefit of matching the behaviour of `iface->get_ip_address()` when not connected.